src/olympia/addons/models.py

```diff
@@ -892,7 +892,7 @@ def initialize_addon_from_upload(cls, data, upload, channel, user):
             )
         if user:
             AddonUser(addon=addon, user=user).save()
-        activity.log_create(amo.LOG.CREATE_ADDON, addon)
+        activity.log_create(amo.LOG.CREATE_ADDON, addon, user=user)
```

Comment:
This is the crucial bit I needed to solve, but getting to that point involved making sure `FileUpload`s always had a `user` set, hence the big refactor

Reply:
"+557 −231" 😿

Reply:
Mostly tests though !